### PR TITLE
Update GitHub Actions workflows

### PR DESCRIPTION
**Pull request by DavisVaughan**
Hopefully switching to pak fixes our failing builds 
https://github.com/tidyverse/dplyr/runs/3398502012

Update:
Turns out switching from xenial to bionic in #5959 required a cache bump to redownload stringi, and that is what was causing the failures. I think that updating the workflows here has also triggered the cache bump, which is why we get green checks now.

Hopefully the pkgdown build works (I've added required packages to Config/Needs/website), we won't know until we merge.

### CODE REVIEW

File: .github/workflows/R-CMD-check.yaml [hunk 1]
```diff
@@ -28,71 +28,40 @@ jobs:
...
+          - {os: windows-latest, r: '3.6', rspm: "https://packagemanager.rstudio.com/cran/latest"}
+          - {os: ubuntu-18.04,   r: 'devel', rspm: "https://packagemanager.rstudio.com/cran/__linux__/bionic/latest", http-user-agent: "R/4.1.0 (ubuntu-18.04) R (4.1.0 x86_64-pc-linux-gnu x86_64 linux-gnu) on GitHub Actions" }
           - {os: ubuntu-18.04,   r: 'release', rspm: "https://packagemanager.rstudio.com/cran/__linux__/bionic/latest"}
           - {os: ubuntu-18.04,   r: 'oldrel',  rspm: "https://packagemanager.rstudio.com/cran/__linux__/bionic/latest"}
           - {os: ubuntu-18.04,   r: '3.6',     rspm: "https://packagemanager.rstudio.com/cran/__linux__/bionic/latest"}
           - {os: ubuntu-18.04,   r: '3.5',     rspm: "https://packagemanager.rstudio.com/cran/__linux__/bionic/latest"}
           - {os: ubuntu-18.04,   r: '3.4',     rspm: "https://packagemanager.rstudio.com/cran/__linux__/bionic/latest"}
-          - {os: ubuntu-18.04,   r: '3.3',     rspm: "https://packagemanager.rstudio.com/cran/__linux__/bionic/latest"}
```
Comment (DavisVaughan) [inline .github/workflows/R-CMD-check.yaml:38]: Do you want to keep 3.3 support? I can add it back in

File: .github/workflows/pkgdown.yaml [hunk 2]
```diff
@@ -1,49 +1,39 @@
...
-          restore-keys: macOS-r-4.0-2-
-
-      - name: Install dependencies
-        run: |
-          remotes::install_cran("tidyverse")
-          remotes::install_cran("shiny")
-          install.packages("remotes")
-          remotes::install_deps(dependencies = TRUE)
-          remotes::install_dev("pkgdown")
-          remotes::install_github("tidyverse/tidytemplate")
```
Comment (DavisVaughan) [inline .github/workflows/pkgdown.yaml:39]: I've moved these to `Config/Needs/website`. Not sure if we really need dev pkgdown or not, but I added it anyways

File: .github/workflows/R-CMD-check.yaml (hunk 1 above)
Comment (romainfrancois) [reply .github/workflows/R-CMD-check.yaml:38]: Might be time to let it go, but in that case perhaps we should also edit here: 

 https://github.com/tidyverse/dplyr/blob/b206089fabda9cb12cca09b366b87090ddaf14f6/DESCRIPTION#L31

**Review by romainfrancois — APPROVED** [review]
Maybe bump to R (>= 3.4) in DESCRIPTION ?

**Pull request merged**